src/main/java/com/google/sps/servlets/startGameServlet.java

```diff
@@ -50,36 +50,28 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Cookie cookies[] = request.getCookies();
     Entity userEntity = UserUtils.getUserFromCookie(cookies, datastore);
  
-    if(userEntity == null){
-        // no user entity, send back to login
-        response.sendRedirect("/index.html");
-        return;
+    String newGame = GameUtils.joinGame(userEntity, gameName, datastore, null);
+ 
+    if(newGame == "noGameName"){
+      response.sendRedirect("/start.html");
+      return;
+    }
+ 
+    if(newGame == "nullUserEntity" || newGame == "createGameFailed" || newGame == "userAddedFailed" || newGame == "gameAddedFailed"){
+      response.sendRedirect("/index.html");
+      return;
     }
  
-    // prevent users from joining more than one game
-    if((String) userEntity.getProperty("gameId") != null){
+    if(newGame == "alreadyHasGame"){
       PrintWriter out = response.getWriter();
       out.println("<p>You are already in a game, see the gameboard here:</p>");
       out.println("<a href = \"gameBoard.html\"><h3>Your game</h3></a>");
       return;
     }
```

Comment:
We could probably just skip this and redirect them right away to the gameboard.

Reply:
That makes a lot more sense; that's changed.